pulpcore/cli/core/repository.py

```diff
@@ -32,3 +47,49 @@ def repository(ctx: click.Context, pulp_ctx: PulpCLIContext) -> None:
 
 repository.add_command(list_command(decorators=filter_options))
 repository.add_command(version_command(decorators=[], list_only=True))
+
+
+@repository.command()
+@pass_repository_context
+@pass_pulp_context
+@repositories_option
+@pulp_option(
+    "--all",
+    "all_repos",
+    is_flag=True,
+    show_default=True,
+    default=False,
+    help=_("Select all repositories."),
+)
+@click.option(
+    "--keep-version",
+    "keep_versions",
+    type=tuple([str, int]),
+    help=_("List of (pulp hrefs) repository versions to keep."),
+    callback=_version_list_callback,
+    multiple=True,
+    required=False,
+)
+def reclaim_space(
+    pulp_ctx: PulpCLIContext,
+    repository_ctx: PulpRepositoryContext,
+    repositories: Iterable[EntityFieldDefinition],
+    all_repos: bool,
+    keep_versions: Iterable[PulpRepositoryVersionContext],
+) -> None:
+    pulp_ctx.needs_plugin(PluginRequirement("core", specifier=">=3.19.0"))
+    if not (repositories or all_repos):
+        raise click.UsageError(_("Either --repository or --all must be specified."))
+    if all_repos:
+        repository_hrefs = ["*"]
+    else:
+        repository_hrefs = [
+            repo.pulp_href for repo in repositories if isinstance(repo, PulpRepositoryContext)
+        ]
+
+    versions_keeplist = [version.pulp_href for version in keep_versions]
+    print(versions_keeplist)
```

Comment:
This is not needed.

Reply:
@mdellweg if you only mean the print, that's left there by accident. Or are you referring to the variables as well?

Reply:
No i mean lines 86 to 91

Reply:
you dont like storing the lists into variables and would prefer passing then straight there, or is there something else I'm missing?

Reply:
No you ought to pass the contexts to the glue layer. No conversion at this level.